builders/xcm/xcm-transactor.md

```diff
@@ -0,0 +1,194 @@
+3. (Optional) Disable/enable the include options slider. This will allow you to query the address authorized for a given index or request all addresses for all registered indexes
+4. If you've enabled the slider, enter the index to query
+5. Send the query by clicking on the **+** button
+
+![Check Registered Derivative Indexes](/images/builders/xcm/xcm-transactor/xcmtransactor-2.png)
+
+## XCM-Transactor Precompile {: #xcmtransactor-precompile}
+
+The XCM-Transactor precompile contract allows developers to access the XCM-Transactor pallet features through the Ethereum API of Moonbeam-based networks. Similar to other [precompile contracts](/builders/build/canonical-contracts/precompiles/){target=_blank}, the XCM-Transactor precompile is located at the following addresses:
+
+=== "Moonbeam"
+     ```
+     {{networks.moonbeam.precompiles.xcm_transactor}}
+     ```
+
+=== "Moonriver"
+     ```
+     {{networks.moonriver.precompiles.xcm_transactor}}
+     ```
+
+=== "Moonbase Alpha"
+     ```
+     {{networks.moonbase.precompiles.xcm_transactor}}
+     ```
+
+### The XCM-Transactor Solidity Interface {: #xcmtrasactor-solidity-interface } 
+
+[XcmTransactor.sol](https://github.com/PureStake/moonbeam/blob/master/precompiles/xcm-transactor/XcmTransactor.sol){target=_blank} is an interface through which developers can interact with the XCM-Transactor pallet using the Ethereum API.
+
+The interface includes the following functions:
+
+ - **index_to_account**(*uint16* index) — read-only function that returns the registered address authorized to operate using a derivative account of the Moonbeam-based network sovereign account for the given index
+  - **transact_info**(*Multilocation* *memory* multilocation) — read-only function that, for a given chain defined as a multilocation, returns the transact information
+  <!-- - **transact_info_with_signed**(*Multilocation* *memory* multilocation) — read-only function that, for a given chain defined as a multilocation, returns the transact information considering the 3 XCM instructions associated to the external call execution, but also returns extra weight information associated with the `decendOrigin` XCM instruction to execute the remote call  -->
+ - **fee_per_second**(*Multilocation* *memory* multilocation) — read-only function that, for a given asset as a multilocation, returns units of token per second of the XCM execution that is charged as the XCM execution fee. This is useful when, for a given chain, there are multiple assets that can be used for fee payment
+ - **transact_through_derivative**(*uint8* transactor, *uint16* index, *address*  address, *uint64* weight, *bytes* *memory* inner_call) — function that represents the `transactThroughDerivative` method described in the [previous example](#xcmtransactor-transact-through-derivative). Instead of the currency ID (asset ID), you'll need to provide the assets precompile address for the `address` of the token that is used for fee payment
```

Comment:
it might be helpful to link back to instructions on how to calculate the assets precompile address